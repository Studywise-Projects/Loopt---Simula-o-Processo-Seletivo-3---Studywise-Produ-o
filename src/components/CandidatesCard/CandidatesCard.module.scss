@import '@/styles/_global.scss';

#cardRoot {
  @media only screen and (max-width: 768px) {
    box-shadow: none;
  }
}

.card {
  border-radius: 20px;
  width: 100%;

  @media only screen and (min-width: 1024px) {
    box-shadow: 4px 4px 15px 0px rgba(0, 0, 0, 0.10);
    width: 65%;
  }
}

.cardContent {
  display: flex;
  background-color: transparent;
  flex-direction: column;
  padding: 16px;
  gap: 16px;

  @media only screen and (min-width: 1024px) {
    background-color: $white;
    border-radius: 20px;
  }
}

.cardCandidate {
  align-items: center;
  box-shadow: 4px 4px 15px 0px rgba(0, 0, 0, 0.10);
  border-radius: 20px;
  padding: 16px;
  flex-direction: row;
  justify-content: space-between;

  @media only screen and (min-width: 1024px) {
    box-shadow: none;
  }
}

.cardCandidate:hover {
  cursor: pointer;
  transform: scale(1.01);
  opacity: 0.7;
}

.cardHighlight {
  align-items: center;
  display: flex;
  gap: 16px;
  justify-content: center;
}

.profilePicture {
  border-radius: 100%;
}

.highlightContent {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  gap: 4px;
  
  @media only screen and (min-width: 768px) {
    justify-content: center;
  }
}

.highlightContentWithButton {
  width: 110px;
}

.skillsContent {
  display: none;

  @media only screen and (min-width: 768px) {
    align-items: flex-start;
    display: flex;
    flex-direction: column;
    gap: 8px;
    justify-content: center;
    width: 180px;
  }
}

.dividerContainer {
  display: none;

  @media only screen and (min-width: 1024px) {
    align-items: center;
    display: flex;
    justify-content: center;
  }
}

.buttonContainer {
  display: flex;
}

.dividerContainer:last-child {
  display: none;
}

.divider {
  display: none;
  
  @media only screen and (min-width: 1024px) {
    align-self: center;
    width: 70%;
  }
}