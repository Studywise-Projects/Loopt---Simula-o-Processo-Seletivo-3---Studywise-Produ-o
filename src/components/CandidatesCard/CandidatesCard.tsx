import { Box, Card, CardContent, Divider, Stack } from '@mui/material';
import Typography from '../Typography/Typography';
import styles from './CandidatesCard.module.scss';
import Image from 'next/image';
import { ICandidate } from '../../interfaces/ICandidate';
import useCandidatesStore from '../../stores/candidates';
import { useRouter } from 'next/router';
import Button from '../Button/Button';
import formatCandidateName from '@/utils/formatCandidateName';
import formatArrayToString from '@/utils/formatArrayToString';
import classNames from 'classnames';

interface ICandidatesCard {
  candidates: Array<ICandidate>;
  withButton?: boolean;
}

function CandidatesCard({ candidates, withButton = false }: ICandidatesCard) {
  const setSelectedCandidate = useCandidatesStore(
    (state) => state.setSelectedCandidate,
  );

  const highlightContentStyles = classNames(styles.highlightContent, {
    [styles.highlightContentWithButton]: withButton === true,
  });

  const router = useRouter();

  return (
    <Card className={styles.card} id={styles.cardRoot}>
      <CardContent className={styles.cardContent}>
        {candidates?.map((candidate: ICandidate) => (
          <>
            <Stack
              data-testid='candidates-card'
              key={candidate.id}
              className={styles.cardCandidate}
              onClick={() => {
                setSelectedCandidate(candidate);
                router.push({
                  pathname: `/candidates/${candidate.id}`,
                  hash:
                    router.asPath === '/candidates'
                      ? router.asPath.replaceAll('/candidates', '')
                      : router.asPath.replaceAll('/candidates/', ''),
                });
                console.log(router.asPath);
              }}
            >
              <Box className={styles.cardHighlight}>
                <Image
                  src={candidate.picture}
                  alt={candidate.name}
                  width={64}
                  height={64}
                  className={styles.profilePicture}
                />
                <Box className={highlightContentStyles}>
                  <Typography
                    variant='subtitle'
                    text={
                      withButton === false
                        ? candidate.name
                        : formatCandidateName(candidate.name)
                    }
                  />
                  <Typography variant='body' text={`${candidate.age} anos`} />
                </Box>
              </Box>
              <Box className={styles.skillsContent}>
                <Typography variant='subtitle2' text='Habilidades' />
                <Typography
                  variant='caption'
                  text={formatArrayToString(candidate.skills, 0, 25)}
                />
              </Box>
              {withButton === true ? (
                <Box className={styles.buttonContainer}>
                  <Button
                    text='Selecionar'
                    isAuxButton={true}
                    handleClick={() => {
                      setSelectedCandidate(candidate);
                      router.push({
                        pathname: `/candidates/${candidate.id}`,
                        hash: router.asPath.replace('/candidates/', ''),
                      });
                    }}
                  />
                </Box>
              ) : (
                <></>
              )}
            </Stack>
            <Box className={styles.dividerContainer}>
              <Divider className={styles.divider} />
            </Box>
          </>
        ))}
      </CardContent>
    </Card>
  );
}

export default CandidatesCard;
