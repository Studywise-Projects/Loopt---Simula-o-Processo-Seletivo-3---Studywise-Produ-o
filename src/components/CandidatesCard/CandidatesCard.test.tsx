import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import CandidatesCard from './CandidatesCard';

const candidates = [
  {
    id: 0,
    name: 'test',
    age: 32,
    jobId: 1,
    picture: '',
    skills: [''],
    approved: false,
  },
];

describe('CandidatesCard', () => {
  it('Should match the snapshot', () => {
    const wrapper = shallow(
      <CandidatesCard
        candidates={[
          {
            id: 0,
            name: 'test',
            age: 32,
            jobId: 1,
            picture: '',
            skills: [''],
            approved: false,
          },
        ]}
        handleClick={() => undefined}
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('Should call handleClickMock function once', () => {
    const handleClickMock = jest.fn();

    const wrapper = shallow(
      <CandidatesCard candidates={candidates} handleClick={handleClickMock} />,
    );

    wrapper.find('[data-testid="candidates-card"]').simulate('click');

    expect(handleClickMock).toBeCalledTimes(1);
  });
});
