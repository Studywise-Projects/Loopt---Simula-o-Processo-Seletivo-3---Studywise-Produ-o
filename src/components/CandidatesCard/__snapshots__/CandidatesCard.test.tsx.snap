// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CandidatesCard Should match the snapshot 1`] = `ShallowWrapper {}`;
